csa.rs

```diff
@@ -0,0 +1,134 @@
+use std::io;
+
+const MAX_STATIONS: usize = 100000;
+
+#[derive(Debug)]
+struct Connection {
+    departure_station: usize,
+    arrival_station: usize,
```

Comment:
Ha, sure, you'll have to `as usize` when using it as a slice index as the implementation is `impl<T> Index<usize> for [T]` and there is no implicit conversion from `u32` to `usize`